src/vcpkg/postbuildlint.cpp

```diff
@@ -1121,73 +1266,87 @@ namespace vcpkg::PostBuildLint
-
-                std::vector<Path> dlls;
-                dlls.insert(dlls.cend(), debug_dlls.cbegin(), debug_dlls.cend());
-                dlls.insert(dlls.cend(), release_dlls.cbegin(), release_dlls.cend());
+                std::vector<Path> libs;
+                libs.insert(libs.cend(), debug_libs.cbegin(), debug_libs.cend());
+                libs.insert(libs.cend(), release_libs.cbegin(), release_libs.cend());
+                error_count += check_lib_architecture(
+                    pre_build_info.target_architecture, pre_build_info.cmake_system_name, libs, fs);
+            }
 
-                if (!toolset.dumpbin.empty() && !build_info.policies.is_enabled(BuildPolicy::SKIP_DUMPBIN_CHECKS))
-                {
-                    error_count += check_exports_of_dlls(build_info.policies, dlls, toolset.dumpbin);
-                    error_count += check_uwp_bit_of_dlls(pre_build_info.cmake_system_name, dlls, toolset.dumpbin);
-                    error_count +=
-                        check_outdated_crt_linkage_of_dlls(dlls, toolset.dumpbin, build_info, pre_build_info);
-                }
+            std::vector<Path> debug_dlls = fs.get_regular_files_recursive(debug_bin_dir, IgnoreErrors{});
+            Util::erase_remove_if(debug_dlls, NotExtensionCaseInsensitive{".dll"});
+            std::vector<Path> release_dlls = fs.get_regular_files_recursive(release_bin_dir, IgnoreErrors{});
+            Util::erase_remove_if(release_dlls, NotExtensionCaseInsensitive{".dll"});
 
-#if defined(_WIN32)
-                error_count += check_dll_architecture(pre_build_info.target_architecture, dlls, fs);
-#endif
-                break;
-            }
-            case LinkageType::STATIC:
+            switch (build_info.library_linkage)
             {
-                auto dlls = release_dlls;
-                dlls.insert(dlls.end(), debug_dlls.begin(), debug_dlls.end());
-                error_count += check_no_dlls_present(build_info.policies, dlls);
+                case LinkageType::DYNAMIC:
+                {
+                    if (!pre_build_info.build_type &&
+                        !build_info.policies.is_enabled(BuildPolicy::MISMATCHED_NUMBER_OF_BINARIES))
+                        error_count += check_matching_debug_and_release_binaries(debug_dlls, release_dlls);
+
+                    error_count += check_lib_files_are_available_if_dlls_are_available(
+                        build_info.policies, debug_libs.size(), debug_dlls.size(), debug_lib_dir);
+                    error_count += check_lib_files_are_available_if_dlls_are_available(
+                        build_info.policies, release_libs.size(), release_dlls.size(), release_lib_dir);
+
+                    std::vector<PostBuildCheckDllData> dlls;
+                    dlls.reserve(debug_dlls.size() + release_dlls.size());
+                    for (const Path& dll : debug_dlls)
+                    {
+                        auto maybe_dll_data = try_load_dll_data(fs, dll);
```

Comment:
Shouldn't failing to load dll data be considered an error as well?

Reply:
I don't think so: we said that post build checks are best effort?

Reply:
Yes, but they're also nonfatal warnings. I think "can't parse the DLL" and "DLL has the wrong directives" are both under the category of "wrong DLL" that is useful to diagnose.